packages/next/src/builders/dev-server/dev-server.impl.ts

```diff
@@ -24,78 +20,84 @@ try {
+    handle(req, res);
+  });
+  return new Promise((resolve, reject) => {
+    server.on('error', (error: Error) => {
+      if (error) {
+        reject(error);
+      }
+    });
+    server.listen(options.port, () => {
+      resolve();
+    });
+  });
+};
 
 function run(
   options: NextBuildBuilderOptions,
   context: BuilderContext
 ): Observable<BuilderOutput> {
   const buildTarget = targetFromTargetString(options.buildTarget);
+  const customServerTarget =
+    options.customServerTarget &&
+    targetFromTargetString(options.customServerTarget);
 
+  const success: BuilderOutput = { success: true };
   const build$ = !options.dev
     ? scheduleTargetAndForget(context, buildTarget)
-    : of({ success: true });
+    : of(success);
+  const customServer$ = customServerTarget
+    ? scheduleTargetAndForget(context, customServerTarget)
+    : of(success);
 
-  return build$.pipe(
-    concatMap(r => {
-      if (!r.success) return of(r);
-      return from(context.getTargetOptions(buildTarget)).pipe(
-        concatMap((buildOptions: any) => {
-          const root = path.resolve(context.workspaceRoot, buildOptions.root);
+  return forkJoin(build$, customServer$).pipe(
+    concatMap(([buildResult, customServerResult]) => {
+      if (!buildResult.success) return of(buildResult);
+      if (!customServerResult.success) return of(customServerResult);
+
+      const customServerEntry = customServerResult.outfile;
 
-          const config = prepareConfig(
+      return from(context.getTargetOptions(buildTarget)).pipe(
+        concatMap((buildOptions: JsonObject) => {
+          const root = path.resolve(
             context.workspaceRoot,
-            buildOptions.root,
-            buildOptions.outputPath,
-            options.dev ? PHASE_DEVELOPMENT_SERVER : PHASE_PRODUCTION_SERVER
+            buildOptions.root as string
           );
 
-          const settings = {
+          const nextApp = next({
             dev: options.dev,
             dir: root,
-            staticMarkup: options.staticMarkup,
-            quiet: options.quiet,
-            conf: config,
-            port: options.port,
-            path: options.customServerPath
-          };
+            quiet: options.quiet
+          });
 
-          const server = options.customServerPath
-            ? customServer
-            : defaultServer;
+          let startServer: StartServerFn;
+          if (customServerEntry) {
+            startServer = require(customServerEntry as string).startServer;
+          } else {
+            startServer = defaultStartServer;
+          }
 
-          return from(server(settings)).pipe(
+          return from(startServer(nextApp, options)).pipe(
             tap(() => {
-              context.logger.info(`Ready on http://localhost:${settings.port}`);
+              context.logger.info(`Ready on http://localhost:${options.port}`);
             }),
             switchMap(
-              e =>
+              _e =>
```

Comment:
On this bit here...is there a reason we don't replace this with simply:

```typescript
() => of({
    baseUrl: `http://localhost:${options.port}`,
    success: true
})
```

? (I know this isn't strictly your code, but while you're here... :)

Reply:
Just tried this out, but the dev server exits right after it starts listening.

I guess the reason is that `of({})` completes the observable immediately, but `new Observable(obs => { obs.next({}); })` creates an observable which emits one value but never completes.

Reply:
D'oh, of course. Thanks for checking.